packages/frontend/component/src/ui/menu/desktop/sub.tsx

```diff
@@ -24,18 +27,33 @@ export const DesktopMenuSub = ({
     suffixIcon: <ArrowRightSmallIcon />,
   });
 
+  const { handleOpenChange, contentOffset, contentRef } =
+    useMenuContentController({
+      defaultOpen,
+      onOpenChange,
+      side: 'right',
+      sideOffset: (sideOffset ?? 0) + 12,
+    });
+
   return (
-    <DropdownMenu.Sub {...subOptions}>
+    <DropdownMenu.Sub
+      defaultOpen={defaultOpen}
+      onOpenChange={handleOpenChange}
+      {...otherSubOptions}
+    >
       <DropdownMenu.SubTrigger className={className} {...otherProps}>
         {children}
       </DropdownMenu.SubTrigger>
       <DropdownMenu.Portal {...portalOptions}>
         <DropdownMenu.SubContent
-          sideOffset={12}
+          sideOffset={contentOffset}
+          ref={contentRef}
           className={useMemo(
             () => clsx(styles.menuContent, subContentClassName),
             [subContentClassName]
           )}
+          style={{ zIndex: 'var(--affine-z-index-popover)', ...contentStyle }}
+          avoidCollisions={false}
```

Comment:
any reason why this does not work?

Reply:
This will only place the menu in the opposite side when there is not enough space. For example, if there is not enough space bottom side, the menu will be displayed top side. But when there is not enough space in both side, it will only go beyond the viewport and will not be displayed in the middle.

Reply:
interesting. I thought https://floating-ui.com/docs/shift will do this?

Reply:
It's same with floating ui shift. But I am used to doing it myself. If there is any issue, we can replace it with float ui